clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/model/ElasticSearchEntityTagsProvider.java

```diff
@@ -496,8 +502,38 @@ public Map reconcile(String cloudProvider, String account, String region, boolea
       results.put("deletedFromElasticsearch", true);
 
       if (deleteFromSource) {
-        Lists.partition(entityTagsForNamespace, 50).forEach(front50Service::batchUpdate);
-        results.put("deletedFromSource", true);
+        ExecutorService executor =
+            Executors.newFixedThreadPool(
+                6,
+                new ThreadFactoryBuilder()
+                    .setNameFormat(ElasticSearchEntityTagsProvider.class.getSimpleName() + "-%d")
+                    .build());
+
+        AtomicLong countProcessed = new AtomicLong();
+
+        Lists.partition(entityTagsForNamespace, 50)
+            .forEach(
+                entityTags ->
+                    executor.submit(
+                        () -> {
```

Comment:
it is probably worth a try/catch around the whole runnable block with some error logging if things fail

Reply:
definitely, thanks!